src/aws_signature.erl

```diff
@@ -340,6 +355,11 @@ maybe_add_session_token_query_param(QueryParams, undefined) ->
 maybe_add_session_token_query_param(QueryParams, SessionToken) ->
     [{<<"X-Amz-Security-Token">>, SessionToken} | QueryParams].
 
+maybe_add_tag_query_param(QueryParams, undefined) ->
+    QueryParams;
+maybe_add_tag_query_param(QueryParams, Tag) ->
+    [{<<"X-Amz-Tagging">>, Tag} | QueryParams].
```

Comment:
From quick search online, it seems to me that `x-amz-tagging` always needs to be send in the headers, so there is no reason to put it in the params.

See https://stackoverflow.com/a/77418874.

Reply:
Yes, if people want to generate a pre-signed URL with tagging, then x-amz-tagging needs to be included. However, we need to support both scenarios, with and without tagging.

Reply:
My understanding is that there is no reason to add x-amz-tagging as query param. It's either sent as header or not. I may totally be missing something though :)

Reply:
Ok, I understand what you mean now. However, we actually need to add x-amz-tagging to the query parameters. Based on the link you shared earlier, the expected output for the pre-signed URL should be:

Presigned URL: https://xxx.s3.eu-west-1.amazonaws.com/xxx.xxx?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=xxx&X-Amz-Date=xxx&X-Amz-Expires=3600&X-Amz-Signature=xxx&X-Amz-SignedHeaders=host%3Bx-amz-tagging&x-id=PutObject

Reply:
I understand what you mean now, and you are correct. It should not be needed.